tf_big/cc/kernels/big_kernels.cc

```diff
@@ -293,11 +312,38 @@ class BigRandomUniformOp : public OpKernel {
     mpz_clear(tmp);
 
     Tensor* res;
-    OP_REQUIRES_OK(ctx, ctx->allocate_output(0, TensorShape{}, &res));
-    res->scalar<Variant>()() = BigTensor(res_matrix);
+    OP_REQUIRES_OK(ctx, ctx->allocate_output(0, shape, &res));
+    res->flat<Variant>()(0) = BigTensor(res_matrix);
   }
 };
 
+// class BigRandomPrimeOp : public OpKernel {
```

Comment:
finish this implementation; see eg [libpaillier keygen](https://github.com/snipsco/paillier-libraries-benchmarks/blob/master/c-libpaillier/libpaillier-0.8/paillier.c#L57-L114)

Reply:
I pushed some code. It's still a WIP. 

Just wanted to confirm that BigRandomPrimeOp should return, p, q and n? and also p, q, n is a tensor based on the input shape, it's won't be a scalar? 

Do we also need to compute [`lcm`](https://github.com/snipsco/paillier-libraries-benchmarks/blob/6a6881b3fdd67b744ce90d7baf11197778584576/c-libpaillier/libpaillier-0.8/paillier.c#L102) between p and q?

Reply:
it's not a bad idea to return all three, but in that case we should rename to something like `BigRandomRsaModulusOp`. `mpz_lcm` and `complete_prvkey` should *not* happen here. all values returned are scalars.

Reply:
Ok sounds good, I will do that. unless you prefer to just return one value. I took this approach because we can generate prime number and checks `n` in one place.

Reply:
Hey @mortendahl, I think it's ready for an initial review. I am now returning scalars. However for some reasons, the tensor shapes I am getting at the end are (1, 1) instead of (). I tried to set the shape to {} [here ](https://github.com/tf-encrypted/tf-big/blob/22ce1a5ad94b275281716ea952b9fa6cefd4a8c7/tf_big/cc/kernels/big_kernels.cc#L363) and [here](https://github.com/tf-encrypted/tf-big/blob/22ce1a5ad94b275281716ea952b9fa6cefd4a8c7/tf_big/cc/ops/big_ops.cc#L45). Maybe because the data is stored in an Eigen matrix.

Reply:
ok with the changes discussed this morning, `random_rsa_modulus` is now returning scalar tensors of shape (). However I realized `convert_from_tensor` convert scalars of shape () to tensors of shape (1, 1) :). I will look into it.

Reply:
ok `convert_from_tensor` is finally returning a scalar. I had to change `ToTensor` a bit.

It should be ready to be reviewed. thanks